clients/src/main/java/org/apache/kafka/common/telemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.telemetry;
+
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetricsReporter} may implement this interface to indicate support for collecting client
+ * telemetry on the client or server side.
```

Comment:
Also, I'm confused if this is implemented on the client, broker, or both? It seems like its two methods are somewhat mutually exclusive, depending on the context (client or broker) in which it is being run.

Reply:
Better to divide the interfaces, I shall do that.

Reply:
Can you elaborate? The JavaDoc comment below clearly say:
```
Implemented by the client {@link MetricsReporter} to provide a {@link ClientTelemetrySender} instance.
```

What is actually a contraction to what in the KIP:
```
A {@link MetricsReporter} may implement this interface to indicate support for collecting client telemetry on the server side
public interface ClientTelemetry {
```

So it seems to be for clients only. Not sure what you mean by "divide the interfaces"? Should there be two separate `ClientTelemetry` one for server one for client?